### ISSUE TITLE
DMA-related assertion failure when working with gasnet memory

### ISSUE THREAD

**Issue by streichler** [StanfordLegion/legion#235]
This has been failing for a while in the gitlab CI runs it seems.

```
cd tutorial/09_custom_mapper
make LG_RT_DIR=../../runtime USE_GASNET=1 DEBUG=1 -j4
mpirun -x GASNET_BACKTRACE=1 -n 2 -H n0000,n0001 ./custom_mapper
```

The test passes about 50% of the time, but the rest of the time fails with bad looking stuff like this:
```
...
Operation timeline for task unnamed_task_1: ready=47692281 start=47712978 stop=48680573
Runtime overhead for task unnamed_task_1: runtime=0.7% wait=0.0%
[1 - 7fdbc8bab700] {4}{dma}: WARNING: gasnet->gasnet copy being serialized on local node (1)
custom_mapper: ../../runtime/realm/operation.cc:166: virtual void Realm::Operation::mark_completed(): Assertion `(prev == Status::RUNNING) || (prev == Status::TERMINATED_EARLY) || (prev == Status::CANCELLED)' failed.
*** Caught a fatal signal: SIGABRT(6) on node 1/2
...
[1] #8  0x00007fdc16e18bf6 in __assert_fail_base (fmt=0x7fdc16f693b8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=assertion@entry=0x1282650 "(prev == Status::RUNNING) || (prev == Status::TERMINATED_EARLY) || (prev == Status::CANCELLED)", file=file@entry=0x1282590 "../../runtime/realm/operation.cc", line=line@entry=166, function=function@entry=0x12829e0 <Realm::Operation::mark_completed()::__PRETTY_FUNCTION__> "virtual void Realm::Operation::mark_completed()") at assert.c:92
[1] #9  0x00007fdc16e18ca2 in __GI___assert_fail (assertion=0x1282650 "(prev == Status::RUNNING) || (prev == Status::TERMINATED_EARLY) || (prev == Status::CANCELLED)", file=0x1282590 "../../runtime/realm/operation.cc", line=166, function=0x12829e0 <Realm::Operation::mark_completed()::__PRETTY_FUNCTION__> "virtual void Realm::Operation::mark_completed()") at assert.c:101
[1] #10 0x00000000010b4599 in Realm::Operation::mark_completed (this=0x7fdbc7c08fb0) at ../../runtime/realm/operation.cc:164
[1] #11 0x00000000010105a8 in Realm::Operation::work_item_finished (this=0x7fdbc7c08fb0, item=0x35a2420, successful=true) at ../../runtime/realm/operation.inl:77
[1] #12 0x000000000101062a in Realm::Operation::AsyncWorkItem::mark_finished (this=0x35a2420, successful=true) at ../../runtime/realm/operation.inl:115
[1] #13 0x00000000010109ad in LegionRuntime::LowLevel::NotifyXferDesCompleteMessage::handle_request (args=...) at ../../runtime/channel.h:1128
[1] #14 0x000000000102bf8e in IncomingShortMessage<LegionRuntime::LowLevel::NotifyXferDesCompleteMessage::RequestArgs, 181, &LegionRuntime::LowLevel::NotifyXferDesCompleteMessage::handle_request, 2>::run_handler (this=0x7fdbc7c095e0) at ../../runtime/activemsg.h:362
...
```

**Comment by jiazhihao** [comment]
The issue should be fixed by commit 344754.

**Comment by streichler** [comment]
It is failing in the gitlab CI builds even with that commit: https://gitlab.com/StanfordLegion/legion/builds/13844526

**Comment by jiazhihao** [comment]
The issue should be solved by commit 383dad6.
